packages/react-storage/src/components/StorageBrowser/views/LocationsView/LocationsView.tsx

```diff
@@ -22,26 +22,24 @@ const {
   Title,
 } = Controls;
 
-const Loading = () => {
-  const [{ isLoading }] = useLocationsData();
-  return isLoading ? <LoadingElement /> : null;
+const Loading = ({ show }: { show: boolean }) => {
+  return show ? <LoadingElement /> : null;
 };
 
-const LocationsMessage = (): React.JSX.Element | null => {
-  const [{ hasError, message }] = useLocationsData();
-  return hasError ? (
+const LocationsMessage = ({
+  show,
+  message,
+}: {
+  show: boolean;
+  message?: string;
+}): React.JSX.Element | null => {
+  return show ? (
     <Message variant="error">{message ?? DEFAULT_ERROR_MESSAGE}</Message>
   ) : null;
 };
 
-const LocationsEmptyMessage = () => {
-  const [{ data, isLoading, hasError }] = useLocationsData();
-  const shouldShowEmptyMessage =
-    data.result.length === 0 && !isLoading && !hasError;
-
-  return shouldShowEmptyMessage ? (
-    <EmptyMessage>No locations to show.</EmptyMessage>
-  ) : null;
+const LocationsEmptyMessage = ({ show }: { show: boolean }) => {
+  return show ? <EmptyMessage>No locations to show.</EmptyMessage> : null;
 };
```

Comment:
The empty message value should be provided to the `Message` component dynamically

Reply:
Can handle this in a follow up